hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/ha/RequestHedgingProxyProvider.java

```diff
@@ -83,14 +83,11 @@ public RequestHedgingInvocationHandler(
       ExecutorService executor = null;
       CompletionService<Object> completionService;
       try {
-        // Optimization : if only 2 proxies are configured and one had failed
-        // over, then we dont need to create a threadpool etc.
         targetProxies.remove(toIgnore);
-        if (targetProxies.size() == 1) {
-          ProxyInfo<T> proxyInfo = targetProxies.values().iterator().next();
-          Object retVal = method.invoke(proxyInfo.proxy, args);
-          successfulProxy = proxyInfo;
-          return retVal;
+
+        ProxyInfo<T> targetProxy;
```

Comment:
ProxyInfo<T> targetProxy = getSuccessfulOrLeftProxy();

no ?

Reply:
I do an assignment and check at the same time in the next line